core/services/commander/main.py

```diff
@@ -204,7 +206,33 @@ async def root() -> Any:
     return HTMLResponse(content=html_content, status_code=200)
 
 
+def setup_ssh() -> None:
+    # store the key in the docker .config volume
+    key_path = "/root/.config/.ssh/"
+    # authorized_keys is in the pi home directory, we just add our key to it
+    authorized_keys_path = "/home/pi/.ssh/authorized_keys"
+
+    try:
+        Path("key_path").mkdir(parents=True, exist_ok=True)
+        # check if id_rsa.pub exists, creates a new one if it doesnt
+        if not os.path.exists(Path(key_path) / "id_rsa.pub"):
+            subprocess.run(shlex.split(f"mkdir -p {key_path}"), check=True)
+            subprocess.run(shlex.split(f"ssh-keygen -t rsa -f {key_path}/id_rsa -q -N ''"), check=True)
+        # add id_rsa.pub to authorized_keys if not there already
+        if not os.path.exists(authorized_keys_path):
+            subprocess.run(shlex.split(f"touch {authorized_keys_path}"), check=True)
+        with open(authorized_keys_path, "r+", encoding="utf-8") as f:
+            with open(Path(key_path) / "id_rsa.pub", encoding="utf-8") as key_file:
+                my_key = key_file.read()
+                if not any(my_key in line for line in f):
+                    f.write(my_key)
```

Comment:
Need to ensure that the file has an empty line before adding it.

Reply:
as [empty lines are ignored](https://www.ibm.com/docs/en/zos/2.5.0?topic=daemon-format-authorized-keys-file), I just added a newline to keep to code short.